File: app/src/main/java/net/myacxy/squinch/settings/debuglog/DebugLogViewModel.java
package net.myacxy.squinch.settings.debuglog;

import android.databinding.ObservableArrayList;

import net.myacxy.squinch.base.ViewModel;
import net.myacxy.squinch.helpers.DataHelper;
import net.myacxy.squinch.helpers.tracking.DebugLogTracker;

public class DebugLogViewModel implements ViewModel, DebugLogTracker.OnDebugLogChangedListener {

    private final DataHelper dataHelper;
    private final DebugLogTracker debugLogTracker;

    private ObservableArrayList<DebugLogEntry> debugLogEntries = new ObservableArrayList<>();

    public DebugLogViewModel(DataHelper dataHelper, DebugLogTracker debugLogTracker) {
        this.dataHelper = dataHelper;
        this.debugLogTracker = debugLogTracker;

        debugLogTracker.setOnDebugLogChangedListener(this);
        debugLogEntries.addAll(dataHelper.getDebugLogEntries());
    }

    public ObservableArrayList<DebugLogEntry> getDebugLogEntries() {
        return debugLogEntries;
    }

    @Override
    public void onEntryAdded(DebugLogEntry debugLogEntry) {
        debugLogEntries.add(debugLogEntry);
    }
}

File: app/src/main/java/net/myacxy/squinch/settings/debuglog/DebugLogViewModel.kt
package net.myacxy.squinch.settings.debuglog

import android.content.SharedPreferences
import android.databinding.ObservableArrayList
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.rxkotlin.subscribeBy
import io.reactivex.schedulers.Schedulers
import net.myacxy.squinch.base.ViewModel
import net.myacxy.squinch.helpers.DataHelper
import timber.log.Timber

class DebugLogViewModel(
        private val dataHelper: DataHelper,
        debugLogSharedPreferences: SharedPreferences
) : ViewModel {
    val debugLogEntries = ObservableArrayList<DebugLogEntry>()
    private var disposable: Disposable? = null

    fun onAttach() {
        // TODO: 01.02.2018 live update
        disposable = dataHelper.debugLogEntries
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeBy(
                        onSuccess = {
                            debugLogEntries.clear()
                            debugLogEntries.addAll(it)
                        },
                        onError = { Timber.e(it) }
                )
    }

    fun onDetach() {
        disposable?.dispose()
    }
}
